math/least_common_multiple.cpp

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright 2020 @author tjgurwara99
+ * @file
+ *
+ * A basic implementation of LCM function
+ */
+
+#include <cassert>
+#include <iostream>
+
+/**
+ * Function for finding greatest common divisor of two numbers.
+ * @params two integers x and y whose gcd we want to find.
+ * @return greatest common divisor of x and y.
+ */
+unsigned int gcd(unsigned int x, unsigned int y) {
```

Comment:
Could you make another pull request in which the functions are templated?

Eg, `gcd` and `lcm` Function inputs can be int, long, long long, signed or unsigned.

```cpp
Template<typename T>
gcd(const T &a, const T &b)
{....}
```

Because it is a recursive function, use of const references gives a faster memory access. The templates

Reply:
@kvedala I have opened another pull request #845. Once that is complete I'll create a new pull request for these changes.